pipenv/project.py

```diff
@@ -628,15 +633,35 @@ def sources(self):
         else:
             return [DEFAULT_SOURCE]
 
+    def find_source(self, source):
+        """given a source, find it.
+
+        source can be a url or an index name.
+        """
+        if not is_valid_url(source):
+            try:
+                source = self.get_source(name=source)
+            except SourceNotFound:
+                source = self.get_source(url=source)
+        else:
+            source = self.get_source(url=source)
+        return source
+
     def get_source(self, name=None, url=None):
-        for source in self.sources:
+        def find_source(sources, name=None, url=None):
             if name:
-                if source.get('name') == name:
-                    return source
-
+                source = [s for s in sources if s.get('name') == name]
             elif url:
-                if source.get('url') in url:
-                    return source
+                source = [s for s in sources if s.get('url') in url]
```

Comment:
I know this was the case before, but is it a good idea to use `in` here, or should we use `startswith` instead?

Reply:
probably, it seemed odd when I was reimplementing it but I kind of copied what was there... I can't really imagine why we'd use `in` vs `startswith`

Reply:
`git blame` shows @kennethreitz and it dates all the way back to the original implementation (v2.8.7!). Maybe no particular reason, just to hammer out something working first.